pkg/costmodel/costmodel_test.go

```diff
@@ -4,8 +4,99 @@ import (
 	"testing"
 
 	"github.com/opencost/opencost/core/pkg/util"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func TestGetGPUCount(t *testing.T) {
+	tests := []struct {
+		name          string
+		node          *v1.Node
+		expectedGPU   *float64
+		expectedVGPU  *float64
+		expectedError bool
+	}{
+		{
+			name: "Standard NVIDIA GPU",
+			node: &v1.Node{
+				Status: v1.NodeStatus{
+					Capacity: v1.ResourceList{
+						"nvidia.com/gpu": resource.MustParse("2"),
+					},
+				},
+			},
+			expectedGPU:  float64Ptr(2),
+			expectedVGPU: float64Ptr(2),
+		},
+		{
+			name: "NVIDIA GPU with GFD - renameByDefault=true",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"nvidia.com/gpu.replicas": "4",
+						"nvidia.com/gpu.count":    "1",
+					},
+				},
+				Status: v1.NodeStatus{
+					Capacity: v1.ResourceList{
+						"nvidia.com/gpu.shared": resource.MustParse("4"),
+					},
+				},
+			},
+			expectedGPU:  float64Ptr(1),
+			expectedVGPU: float64Ptr(4),
+		},
+		{
+			name: "NVIDIA GPU with GFD - renameByDefault=false",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"nvidia.com/gpu.replicas": "4",
+						"nvidia.com/gpu.count":    "1",
+					},
+				},
+				Status: v1.NodeStatus{
+					Capacity: v1.ResourceList{
+						"nvidia.com/gpu": resource.MustParse("4"),
+					},
+				},
+			},
+			expectedGPU:  float64Ptr(1),
+			expectedVGPU: float64Ptr(4),
+		},
+		{
+			name: "No GPU",
+			node: &v1.Node{
+				Status: v1.NodeStatus{
+					Capacity: v1.ResourceList{},
+				},
+			},
+			expectedGPU:  nil,
+			expectedVGPU: nil,
```

Comment:
these tests are very enlightening, thank you. would you consider adding a case for aws VGPU so I can see what you are expecting there?

Reply:
Will do! For testing the AWS VGPU case I'll need to create more mocks.

Reply:
@ameijer Actually I'm going to hold from implementing tests for the AWS vGPU. It's not clear to me whether it's still widely used within this repo https://github.com/awslabs/aws-virtual-gpu-device-plugin. I also don't have much experience in this. Are you ok if we revisit this at a later point? cc @chipzoller

Reply:
Skip AWS vGPU as no one is using it any more.